daemons/pacemakerd/pacemakerd.c

```diff
@@ -186,21 +186,40 @@ remove_core_file_limit(void)
 {
     struct rlimit cores;
 
-    if (getrlimit(RLIMIT_CORE, &cores) == 0) {
-        if ((cores.rlim_max == 0) && (geteuid() == 0)) {
-            cores.rlim_max = RLIM_INFINITY;
-        } else {
-            crm_trace("Maximum core file size is %llu bytes",
-                      (unsigned long long) cores.rlim_max);
+    // Get current limits
+    if (getrlimit(RLIMIT_CORE, &cores) < 0) {
+        crm_notice("Unable to check system core file limits "
+                   "(consider ensuring the size is unlimited): %s",
+                   strerror(errno));
+        return;
+    }
+
+    // Check whether core dumps are disabled
+    if (cores.rlim_max == 0) {
+        if (geteuid() != 0) { // Yes, and there's nothing we can do about it
+            crm_notice("Core dumps are disabled (consider enabling them)");
+            return;
         }
+        cores.rlim_max = RLIM_INFINITY; // Yes, but we're root, so enable them
+    }
+
+    // Raise soft limit to hard limit (if not already done)
+    if (cores.rlim_cur != cores.rlim_max) {
         cores.rlim_cur = cores.rlim_max;
-        if (setrlimit(RLIMIT_CORE, &cores) == 0) {
+        if (setrlimit(RLIMIT_CORE, &cores) < 0) {
+            crm_notice("Unable to raise system limit on core file size "
+                       "(consider doing so manually): %s",
+                       strerror(errno));
             return;
         }
     }
-    crm_notice("Cannot raise system limit on core file size "
-               "(consider doing so manually): %s",
-               strerror(errno));
+
+    if (cores.rlim_max == RLIM_INFINITY) {
```

Comment:
Current code ensures that rlim_max == rlim_cur or we error out.
If some day we introduce some additional limit via pacemaker-configuration somehow (env-variable e.g. as we may not want cib-access for those basic things) this will write the wrong value.
Going with rlim_cur should be safer thus.

Reply:
Even checking for the disabled case would make sense in that context but atm it would probably raise questions.

Reply:
makes sense